tests/conditional_compilation/conftest.py

```diff
@@ -86,6 +84,25 @@ def pytest_generate_tests(metafunc):
     metafunc.parametrize("test_id, model", params, ids=ids)
 
 
+@pytest.fixture(scope="function")
+def test_info(request, pytestconfig):
+    """Fixture function for getting the additional attributes of the current test."""
+    setattr(request.node._request, "test_info", {})
+    if not hasattr(pytestconfig, "session_info"):
+        setattr(pytestconfig, "session_info", [])
+
+    yield request.node._request.test_info
+
+    pytestconfig.session_info.append(request.node._request.test_info)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def save_session_info(pytestconfig):
+    """Fixture function for saving additional attributes to configuration file."""
+    yield
+    write_test_info(data=pytestconfig.session_info)
```

Comment:
looks like `write_test_info` should be also renamed according to new fixture name